docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -1,40 +1,533 @@
+
+#include_code public_state_struct_write /yarn-project/aztec-nr/aztec/src/state_vars/public_state.nr rust
+
+#### Writing to our `admin` example
+
+#include_code write_admin /yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+#### Writing to our `minters` example
+
+#include_code write_minter /yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+## Private State Variables
+In contrast to public state, private state is persistent state that is **not** visible to the whole world. Depending on the logic of the smart contract, a private state variable's current value will only be known to one entity, or a closed group of entities.
+
+The value of a private state variable can either be shared via an [encrypted log](./events.md#encrypted-events), or offchain via web2, or completely offline: it's up to the app developer.
+
+<!-- TODO: insert link in italics above -->
+
+Aztec private state follows a utxo-based model. That is, a private state's current value is represented as one or many [notes](#notes). Each note is stored as an individual leaf in a utxo-based merkle tree: the [private state tree](./../../../concepts/foundation/state_model.md).
+
+<!-- TODO: insert link in italics above -->
+
+To greatly simplify the experience of writing private state, Aztec.nr provides three different types of private state variable:
+
+- [Singleton<NoteType\>](#singletonnotetype)
+- [ImmutableSingleton<NoteType\>](#immutablesingletonnotetype)
+- [Set<NoteType\>](#setnotetype)
+
+These three structs abstract-away many of Aztec's protocol complexities, by providing intuitive methods to modify notes in the utxo tree in a privacy-preserving way.
+
+:::info
+Note that an app can also choose to emit data via unencrypted log, or to define a note whose data is easy to figure out, then the information is technically not private and could be visible to anyone.
+:::
+
+### Notes 
+Unlike public state variables, which can be arbitrary types, private state variables operate on `NoteType`.
+
+Notes are the fundamental elements in the private world.
+
+A note should conform to the following interface:
+
+#include_code NoteInterface /yarn-project/aztec-nr/aztec/src/note/note_interface.nr rust
+
+The interplay between a private state variable and its notes can be confusing. Here's a summary to aid intuition:
+
+- A private state variable (of type `Singleton`, `ImmutableSingleton` or `Set`) may be declared in [Storage](storage).
+- Every note contains (as a 'header') the contract address and storage slot of the state variable to which it "belongs". A note is said to "belong" to a private state if the storage slot of the private state matches the storage slot contained in the note's header.
```

Comment:
Do you think it would be worth elaborating on what's in the note header?

Reply:
Probably a good idea. Will elaborate more.